src/test/java/io/confluent/examples/streams/GenericAvroIntegrationTest.java

```diff
@@ -128,11 +130,12 @@ public void shouldRoundTripGenericAvroDataThroughKafka() throws Exception {
         outputTopic,
         topologyTestDriver,
         new IntegrationTestUtils.NothingSerde<>(),
-        new KafkaAvroDeserializer(mockSchemaRegistryClient)
+        new KafkaAvroDeserializer(schemaRegistryClient)
       ).stream().map(kv -> (GenericRecord) kv.value).collect(Collectors.toList());
       assertThat(actualValues).isEqualTo(inputValues);
     } finally {
       topologyTestDriver.close();
+      MockSchemaRegistry.dropScope(SCHEMA_REGISTRY_SCOPE);
```

Comment:
To not leak memory, you can drop the scope when you're done with it, so it can get garbage collected. But if you forget, it's not too bad, as it only pollutes the JVM heap while the tests are running.